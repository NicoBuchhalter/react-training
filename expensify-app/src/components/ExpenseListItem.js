import React from 'react';

export default ({ description, amount, createdAt}) => (
  <div>
    <h3>{description}</h3>
    <p>Amount: $ {parseFloat(amount / 100)}</p>
    <p>Created at: {createdAt}</p>
  </div>
); 