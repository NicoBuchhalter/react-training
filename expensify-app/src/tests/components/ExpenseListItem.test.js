import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should render expense item with given data', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
  expect(wrapper).toMatchSnapshot();
});