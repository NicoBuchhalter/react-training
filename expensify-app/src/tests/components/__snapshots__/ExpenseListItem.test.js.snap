// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense item with given data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    Amount: $ 
    7.5
  </p>
  <p>
    Created at: 
    0
  </p>
</div>
`;
